nav2_rviz_plugins/include/nav2_rviz_plugins/goal_tool.hpp

```diff
@@ -50,8 +51,12 @@ class RVIZ_DEFAULT_PLUGINS_PUBLIC GoalTool : public rviz_default_plugins::tools:
 protected:
   void onPoseSet(double x, double y, double theta) override;
 
+private Q_SLOTS:
+  void updateTopic();
+
 private:
-  std::unique_ptr<NavigationDialog> navigation_dialog_;
+  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr publisher_;
+  std::unique_ptr<rviz_common::properties::StringProperty> topic_property_;
```

Comment:
Why does this need to be a pointer?

Reply:
I was just carrying over this code form the InitalPoseTool in from `rviz_default_plugins`.

Reply:
OK, far be it for me to argue with the rviz folks